tsfresh/feature_extraction/settings.py

```diff
@@ -117,7 +119,7 @@ def __init__(self):
             "ar_coefficient": [{"coeff": coeff, "k": k} for coeff in range(5) for k in [10]],
             "mean_abs_change_quantiles": [{"ql": ql, "qh": qh}
                                           for ql in [0., .2, .4, .6, .8] for qh in [.2, .4, .6, .8, 1.]],
-            "fft_coefficient": [{"coeff": coeff} for coeff in range(0, 10)],
+            "fft_coefficient": [{"coeff": k, "attr": a} for a, k in product(["real", "imag"], range(100))],
```

Comment:
yes. I had the same idea. get the maximal length of the time series, and then use that to calculate max coeff. I thought long about this, but I do not think there is a good way to realize that.

We could add a function in the future that infers the extraction settings from a given dataset.

But, I am reading in the details of the DFT. I am not sure yet, but from what I remember the higher order coefficients correspond to the energy in higher frequencies. As most time series suffer from noise, Im expect higher order coefficients not to be too relevant.